src/problem.cpp

```diff
@@ -111,7 +120,8 @@ Problem::FindSolutions(Handle& handle, const FindOptions& options, std::size_t m
             SetTensor(handle, descriptor, buffer.get(), &zero);
         });
 
-        buffers.emplace(pair.first, std::move(buffer));
+        buffers.emplace(pair.first, buffer.get());
```

Comment:
Why does this one (owned buffer) not require TensorID as key v/s other one does?